master: add info about tidb_opt_enable_alternative_logical_plans (#23218)
pingcap/docs: system-variables.md

- Default value: `OFF`
- This variable controls whether the optimizer applies the [`NO_DECORRELATE()`](/optimizer-hints.md#no_decorrelate) hint for all queries that contain a subquery in the `SELECT` list.

> ### `tidb_opt_enable_alternative_logical_plans` <span class="version-mark">Introduced since v8.5.7 and v9.0.0</span>

Author: ```suggestion
### tidb_opt_enable_alternative_logical_plans <span class="version-mark">New in v8.5.7 and v9.0.0</span>
```

- Scope: SESSION | GLOBAL
- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): Yes
- Type: Boolean
- Default value: `OFF`
- This variable controls whether the optimizer additionally builds a logical candidate plan that does not decorrelate in the [correlated subquery decorrelation](/correlated-subquery-optimization.md) scenario.
    - By default, TiDB prioritizes attempting decorrelation rewrites for correlated subqueries.

>     - After you enable this variable, if the decorrelated candidate plan does not generate an equivalent `IndexJoin` candidate with the same access direction as the original correlated subquery, the optimizer additionally retains a candidate plan that does not decorrelate. The optimizer evaluates both the decorrelated and non-decorrelated candidate plans, and selects the [execution plan](/explain-subqueries.md) with the lower cost.

Author: ```suggestion
    - After you enable this variable, if the decorrelated candidate plan fails to generate an equivalent `IndexJoin` candidate plan with the same access direction as the original correlated subquery, the optimizer additionally retains a non-decorrelated candidate plan, evaluates both the decorrelated and non-decorrelated candidate plans, and selects the [execution plan](/explain-subqueries.md) with the lower cost.
```
